clientdb/account.go

```diff
@@ -156,9 +181,15 @@ func readAccount(sourceBucket *bbolt.Bucket,
 }
 
 func serializeAccount(w *bytes.Buffer, a *account.Account) error {
+	rawState := a.State
+	accountIsVersioned := a.Version > account.VersionInitialNoVersion
+	if accountIsVersioned {
+		rawState = setVersionBit(rawState)
```

Comment:
No, the version wasn't stored before. So if we read an existing account, we don't know whether to read an additional byte for the version or not. And because the accounts are embedded within the same data blob for batch snapshots, we also can't use the EOF trick. We could use a separate bucket with additional fields. But that just makes it way more complicated as we'd need to treat the normal accounts and those embedded within the batch snapshots differently.
But I am going to add TLV for all new fields, so adding new ones will become easier.